src/server.rs

```diff
@@ -0,0 +1,119 @@
+        &self,
+        request: Request<RestartDeploymentRequest>,
+    ) -> Result<Response<ResponseMessage>, Status> {
+        let project_uuid = request.into_inner().project_uuid;
+        // TODO: Implement the logic for restarting deployment
+
+        let response = ResponseMessage {
+            // Fill in the fields of the response message
+            // This one is a template for test purposes
+            message: format!("Restart Deployment ! UUID : {}!", project_uuid).into(),
+        };
+        Ok(Response::new(response))
+    }
+
+    async fn delete_deployment(
+        &self,
+        request: Request<DeleteDeploymentRequest>,
+    ) -> Result<Response<ResponseMessage>, Status> {
+        let project_uuid = request.into_inner().project_uuid;
+        // TODO: Implement the logic for deleting deployment
+
+        let response = ResponseMessage {
+            // Fill in the fields of the response message
+            // This one is a template for test purposes
+            message: format!("Delete Deployment ! UUID : {}!", project_uuid).into(),
+        };
+        Ok(Response::new(response))
+    }
+
+    async fn stop_deployment(
+        &self,
+        request: Request<StopDeploymentRequest>,
+    ) -> Result<Response<ResponseMessage>, Status> {
+        let project_uuid = request.into_inner().project_uuid;
+        // TODO: Implement the logic for stopping deployment
+
+        let response = ResponseMessage {
+            // Fill in the fields of the response message
+            // This one is a template for test purposes
+            message: format!("Stop Deployment ! UUID : {}!", project_uuid).into(),
+        };
+        Ok(Response::new(response))
+    }
+
+    async fn deployment_status(
+        &self,
+        request: Request<DeploymentStatusRequest>,
+    ) -> Result<Response<ResponseMessage>, Status> {
+        let project_uuid = request.into_inner().project_uuid;
+        // TODO: Implement the logic for stopping deployment
+
+        let response = ResponseMessage {
+            // Fill in the fields of the response message
+            // This one is a template for test purposes
+            message: format!("Deployment status ! UUID : {}!", project_uuid).into(),
+        };
+        Ok(Response::new(response))
+    }
+
+    async fn apply_config_deployment(
+        &self,
+        request: Request<ApplyConfigDeploymentRequest>,
+    ) -> Result<Response<ResponseMessage>, Status> {
+        let config = request.into_inner().config;
+        // TODO: Implement the logic for applying configuration to the deployment
+
+        let response = ResponseMessage {
+            // Fill in the fields of the response message
+            // This one is a template for test purposes
+            message: format!("Apply Deployment Config ! Config : {}!", config).into(),
+        };
+        Ok(Response::new(response))
+    }
+}
+
+pub(crate) async fn start_server() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
No need to have a specific visibility scope imho.

```suggestion
pub async fn start_server() -> Result<(), Box<dyn std::error::Error>> {
```